client/adventureMap/AdventureMapShortcuts.cpp

```diff
@@ -457,6 +463,72 @@ void AdventureMapShortcuts::zoom( int distance)
 	owner.hotkeyZoom(distance, false);
 }
 
+void AdventureMapShortcuts::search(bool next)
+{
+	// get all relevant objects
+	std::vector<ObjectInstanceID> visitableObjInstances;
+	int3 mapSizes = LOCPLINT->cb->getMapSize();
+	for(int x = 0; x < mapSizes.x; x++)
+		for(int y = 0; y < mapSizes.y; y++)
+			for(int z = 0; z < mapSizes.z; z++)
+				for(auto & obj : LOCPLINT->cb->getVisitableObjs(int3(x, y, z), false))
+					if(obj->ID != MapObjectID::MONSTER && obj->ID != MapObjectID::EVENT && obj->ID != MapObjectID::HERO)
+						visitableObjInstances.push_back(obj->id);
+
+	// count of elements for each group
+	std::map<std::string, int> mapObjCount;
+	for(auto & obj : visitableObjInstances)
+		mapObjCount[{ VLC->objtypeh->getObjectName(LOCPLINT->cb->getObjInstance(obj)->getObjGroupIndex(), LOCPLINT->cb->getObjInstance(obj)->getObjTypeIndex()) }]++;
```

Comment:
Avoid long and complex lines
```suggestion
	// count of elements for each group
	std::map<std::string, int> mapObjCount;
	for(auto & obj : visitableObjInstances)
	{
		const auto * objectPtr = LOCPLINT->cb->getObjInstance(obj);
		const std::string & objectName = VLC->objtypeh->getObjectName(objectPtr->getObjGroupIndex(), objectPtr->getObjTypeIndex());
		mapObjCount[objectName]++;
	}
```